pkg/util/workerqueue/workerqueue.go

```diff
@@ -106,15 +115,56 @@ func (wq *WorkerQueue) processNextWorkItem() bool {
 }
 
 // Run the WorkerQueue processing via the Handler. Will block until stop is closed.
-// Runs threadiness number workers to process the rate limited queue
-func (wq *WorkerQueue) Run(threadiness int, stop <-chan struct{}) {
-	defer wq.queue.ShutDown()
-
-	wq.logger.WithField("threadiness", threadiness).Info("Starting workers...")
-	for i := 0; i < threadiness; i++ {
-		go wait.Until(wq.runWorker, time.Second, stop)
+// Runs a certain number workers to process the rate limited queue
+func (wq *WorkerQueue) Run(workers int, stop <-chan struct{}) {
+	wq.setWorkerCount(workers)
+	wq.logger.WithField("workers", workers).Info("Starting workers...")
+	for i := 0; i < workers; i++ {
+		go wq.run(stop)
 	}
 
 	<-stop
 	wq.logger.Info("...shutting down workers")
+	wq.queue.ShutDown()
+}
+
+func (wq *WorkerQueue) run(stop <-chan struct{}) {
+	wq.inc()
+	wait.Until(wq.runWorker, workFx, stop)
+	wq.dec()
+
+}
+
+// Healthy reports whether all the worker goroutines are running.
+func (wq *WorkerQueue) Healthy() bool {
```

Comment:
```go
wq.mu.Lock()
defer wq.mu.Unlock()
return wq.workers == wq.running
```
?
In this case, this shows why I prefer a lock/defer unlock pattern, as once you past it, it shows much easier exactly what the logic of the function is. Otherwise you have to mentally unwind what `want` and `got` actually mean. In this case it's pretty easy, but be good to set a standard. WDYT?